nemo_asr/calc_rtf.py

```diff
@@ -135,70 +96,61 @@ def main():
         asr_model = ASRModel.from_pretrained(MODEL)
 
     asr_model.to(DEVICE)
-    asr_model.encoder.eval()
-    asr_model.encoder.freeze()
+    asr_model.eval()
     asr_model._prepare_for_export()
 
-    processor = extract_preprocessor(asr_model, DEVICE)
-
-    input_example, input_example_length  = get_samples(WAV, audio_maxlen)
-    logging.info(f'processed example shape: {input_example.shape}')
-    logging.info(f'processed example length shape: {input_example_length}')
-    processed_example, processed_example_length = preprocess_audio(processor, input_example, DEVICE)
-    processed_example = processed_example.repeat(batch_size, 1, 1)
-    processed_example_length = processed_example_length.repeat(batch_size)
-    logging.info(f'processed example shape: {processed_example.size()}')
-    logging.info(f'processed example length shape: {processed_example_length.size()}')
-
-    profiling_context = nullcontext()
-    # if FP16:
-    if PRECISION == '16':
-        precision_context = torch.cuda.amp.autocast()
-    elif PRECISION == 'bf16':
-        precision_context = torch.cuda.amp.autocast(dtype=torch.bfloat16)
-    elif PRECISION == '32':
-        pass
-    else:
-        logging.error(f'unknown precision: {PRECISION}')
-        sys.exit(1)
-
-
-    if decoding_type == 'ctc':
-        asr_model.change_decoding_strategy(decoding_cfg=None)
+    input_example, input_example_length  = get_samples(WAV, total_audio_len)
+    input_example = torch.tensor(input_example).to(DEVICE)
+    input_example = input_example.repeat(batch_size, 1)
     
+
     logging.info(f"running {nbatches} batches; with {warmup_batches} batches warmup; batch_size: {batch_size}")
-    rtfs=[]
+    rtfxs=[]
     for i in range(3): # average over 3 runs
         total_time = 0
-        with profiling_context:
-            with precision_context:
-                with torch.no_grad():
-                    for i in tqdm(range(nbatches + warmup_batches)):
-
+        with torch.cuda.amp.autocast(dtype=torch.bfloat16):
+            with torch.no_grad():
+                for i in tqdm(range(nbatches + warmup_batches), desc='Calculating RTF', unit='batch'):
+                    for idx in range(0, total_audio_len, chunk_len): # process audio in chunks
+                        chunk_singal = input_example[:, idx*SAMPLING_RATE:(idx+chunk_len)*SAMPLING_RATE]
+                        chunk_signal_length = torch.tensor(chunk_len * SAMPLING_RATE).to(DEVICE).repeat(batch_size)
```

Comment:
You're putting a synchronous operation into a hot path of your code here FYI. Can be bad for performance.

Reply:
Do you recommend through iterator? However measurement of duration is outside it.

Reply:
I would say don't worry about it for now. I just wanted to alert you to it as an "FYI" sort of thing.